node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeAllocation.java

```diff
@@ -189,10 +189,13 @@ private boolean offeredNodeHasParentHostnameAlreadyAccepted(NodeCandidate candid
     private boolean violatesExclusivity(NodeCandidate candidate) {
         if (candidate.parentHostname().isEmpty()) return false;
 
-        // In dynamic provisioned zones a node requiring exclusivity must be on a host that has exclusiveTo equal to its owner
-        if (nodeRepository.zone().getCloud().dynamicProvisioning())
-            return requestedNodes.isExclusive() &&
-                    ! candidate.parent.flatMap(Node::exclusiveTo).map(application::equals).orElse(false);
+        // In dynamic provisioned zones, exclusivity is violated if...
+        if (nodeRepository.zone().getCloud().dynamicProvisioning()) {
+                    // If either the parent is dedicated to a cluster type different from this cluster
+            return  ! candidate.parent.flatMap(Node::exclusiveToClusterType).map(cluster.type()::equals).orElse(true) ||
```

Comment:
This first part of the || condition will never actually be true given the other changes in this PR, agree?

Reply:
Yes, this should only happen if `exclusiveToClusterType` somehow changed later, with this PR this is only possible if operator patches the node via /nodes/v2